src/backend/distributed/planner/function_call_delegation.c

```diff
@@ -217,10 +269,29 @@ TryToDelegateFunctionCall(DistributedPlanningContext *planContext)
 	{
 		/*
 		 * We are planning a call to a distributed function within a Citus backend,
-		 * that means that this is the delegated call.
+		 * that means that this is the delegated call. If the function is forcefully
+		 * delegated, capture the distribution argument.
 		 */
-		InDelegatedFunctionCall = true;
-		return NULL;
+		CheckDelegatedFunctionExecution(procedure, funcExpr);
```

Comment:
Maybe 
```c
if (procedure->forceDelegation)
{
	CheckDelegatedFunctionExecution(procedure, funcExpr);
    return NULL;
}
else
{
    /* function calls in a transaction block should not be pushed down unless forced */
    Assert(!IsMultiStatementTransaction());

	InSafeDelegatedFunctionCall = true;
	return NULL;
}
```

Reply:
I've reformatted the code. The only caveat in the above code is, the below will fail when run **without BEGIN/END**

```
CREATE FUNCTION update_data_nonlocal(a integer)
RETURNS void LANGUAGE plpgsql AS $fn$
BEGIN
        UPDATE forcepushdown_schema.test_forcepushdown SET data = 'non-default';
END;
$fn$;

SELECT create_distributed_function(
  'update_data_nonlocal(int)', 'a',
  colocate_with := 'test_forcepushdown',
  force_delegation := true
);



```

Reply:
Please keep in mind
if (procedure->forceDelegation) it has two scenarios 
1) No Begin/End safe
2) Begin/End unsafe 

If we want to keep it simple and fail in both the cases, I can remove the IsMultiStatementTransaction() check for "force-delagation", only caveat is, it will fail unnecessarily.

Reply:
BTW, this assertion is not valid for nested calls
Assert(!IsMultiStatementTransaction());

A simple non-force function calling another function will hit this assertion